gadget/mountedfilesystem.go

```diff
@@ -375,7 +378,53 @@ func (f *mountedFilesystemUpdater) Update() error {
 		}
 	}
 
-	if skipped == len(f.ps.ResolvedContent) {
+	knownContent := make(map[string]bool)
+	for _, c := range f.ps.VolumeStructure.Content {
+		dst := c.Target
+		if strings.HasSuffix(dst, "/") {
+			dst = filepath.Join(dst, filepath.Base(c.UnresolvedSource))
+		}
+		knownContent[dst] = true
+	}
```

Comment:
This is repeated now in 3 places, probably worth creating the map in an ancillary method

Reply:
I have factored it.